contrib/codegen/templates/sets/sets.gotmpl

```diff
@@ -240,7 +240,183 @@ func (s *{{ $kindLowerCamel }}Set) Clone() {{ $resource.Kind }}Set {
 	if s == nil {
 		return nil
 	}
-	return &{{ $kindLowerCamel }}Set{set: sksets.NewResourceSet(s.Generic().Clone().List()...)}
+	return &{{ $kindLowerCamel }}MergedSet{sets: []sksets.ResourceSet{s.Generic()}}
```

Comment:
This is the main effective change, clone now produces a merged set, which is a a list of sets which is iterated through on methods like `Find` or `List`